control/tests/test_api.py

```diff
@@ -56,7 +72,52 @@ def test_no_access_to_theme_for_deleted_control():
     theme = factories.ThemeFactory()
     user = utils.make_audited_user(theme.questionnaire.control)
     theme.questionnaire.control.delete()
-    assert get_theme(user, theme.id).status_code == 404
+    assert update_theme(user, make_update_theme_payload(theme)).status_code == 404
+
+
+def test_cannot_list_themes():
```

Comment:
Here, if I'm not mistaken, we are testing the fact that ListModelMixin is not used anymore for building the API viewset class.

Do we really need to test that `ListModelMixin` is not exposed ? 

- Pro : One reason why I think it's a good idea to do it : because we could consider that exposing that List was actually a bug. If we are solving a bug, then we should have a test.
- Cons : it just adds more tests for something that does not really need to be tested if we consider that we trust that Django Rest Framework does the right thing if we select the right Mixin for the API viewet. My take is that we should avoid adding tests if we can. As a new comer to the code, when you read this test, you could wonder why it's event there.

Reply:
I'm in favor of testing here, because we need to guarantee that the hole is closed. Ignoring things that we don't need is exactly what caused the holes in the first place. 
Listing is forbidden on purpose, and we want to actively keep it that way unless we have a reason to expose it. If in the future we add it by mistake (because we use generics too widely for example), this test will break, which is a good thing.

I also don't trust in general that Django Rest Framework does what we think it does : there are multiple ways to use it (permissions vs querysets for example) and we do not have a global view of what we do with it. 
Though it's true that in this case it seems less likely that a mistake was made... 

Overall, I vote for heavy testing : I know it slows development and future changes, but I think it's a good thing.